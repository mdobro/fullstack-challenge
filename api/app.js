let express = require('express')
let cors = require('cors')
let app = express()

app.use(cors())
app.use(express.json())

//add routers
let contacts = require('./router/contacts')
app.use('/contacts', contacts)

console.log("Starting server on port 3000")

app.listen(3000)