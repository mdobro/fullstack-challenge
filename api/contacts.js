var express = require('express')
var router = express.Router()
var MongoClient = require('mongodb').MongoClient


router.get('/:id', (req, res) => {
  //unpack req
  let id = req.params.id
  //query db
  MongoClient.connect('mongodb://localhost:27017', function (err, client) {
    if (err) throw err

    var db = client.db('ggg')


  })
})

router.post('/', (req, res) => {
  //unpack req
  res.send(req.body)
})

module.exports = router
