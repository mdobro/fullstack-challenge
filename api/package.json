{
  "name": "api",
  "version": "1.0.0",
  "description": "",
  "main": "index.js",
  "scripts": {
    "start": "node app.js",
    "test": "mocha"
  },
  "author": "",
  "license": "ISC",
  "dependencies": {
    "cors": "^2.8.5",
    "express": "^4.16.4",
    "mocha": "^6.0.2",
    "mongodb": "^3.1.13",
    "paperwork": "^2.1.1",
    "supertest": "^4.0.2"
  },
  "devDependencies": {
    "eslint": "^5.15.2"
  }
}
