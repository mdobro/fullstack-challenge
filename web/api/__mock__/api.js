module.exports = {
  getContacts: async function () {
    return [
      {
        first_name: "Test",
        last_name: "Tester",
        phone_number: "111-222-3333",
        email: "test@email.com",
        address: "123 Easy St"
      }
    ]
  },

  postContact: async function (data) {
    return {
      ok: 1,
      n: data.length
    }
  }
}