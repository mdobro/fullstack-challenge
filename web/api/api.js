import fetch from 'node-fetch'

module.exports = {
  getContacts: async function () {
    let res = await fetch('http://localhost:3000/contacts')
    return await res.json()
  },

  postContact: async function (data) {
    let res = await fetch('http://localhost:3000/contacts', {
      method: 'POST',
      headers: new Headers({
        'Content-Type': 'application/json'
      }),
      body: JSON.stringify(data)
    })
    return await res.json()
  }
}