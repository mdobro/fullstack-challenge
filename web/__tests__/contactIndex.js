import React from 'react'
import ContactsScreen from '../pages/contacts/index'
import renderer from 'react-test-renderer';
import api from '../api/api'
jest.mock('../api/api')

test('snapshot of contactsScreen', (done) => {
  api.getContacts()
  .then(res => {
    const component = renderer.create(<ContactsScreen contacts={res}/>,);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    done()
  })
});