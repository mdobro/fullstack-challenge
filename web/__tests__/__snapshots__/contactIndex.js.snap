// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot of contactsScreen 1`] = `
<div>
  <h1>
    Contacts
  </h1>
  <button
    onClick={[Function]}
  >
    Create new Contact
  </button>
  <div
    style={
      Object {
        "marginTop": 10,
      }
    }
  >
    <div
      style={
        Object {
          "borderStyle": "double",
          "marginBottom": 10,
        }
      }
    >
      <h1>
        Test
         
        Tester
      </h1>
      <ul>
        <li>
          Phone Number: 
          111-222-3333
        </li>
        <li>
          Email: 
          test@email.com
        </li>
        <li>
          Address: 
          123 Easy St
        </li>
      </ul>
    </div>
  </div>
</div>
`;
